extensions/xcode/src/components/xcode-simulator-application-list/xcode-simulator-application-list-item.component.tsx

```diff
@@ -95,6 +95,47 @@ export function XcodeSimulatorApplicationListItem(props: { application: XcodeSim
               }
               shortcut={{ modifiers: ["cmd"], key: "backspace" }}
             />
+            <Action
+              icon={Icon.Trash}
+              title="Delete App Data"
+              style={Action.Style.Destructive}
+              onAction={() =>
+                confirmAlert({
+                  title: "Delete App Data",
+                  message:
+                    "This will delete app container folder and app group folder without uninstalling. Are you sure you want to continue?",
+                  primaryAction: {
+                    title: "Delete",
+                    style: Alert.ActionStyle.Destructive,
+                    onAction: async () => {
+                      // Terminate the app if it's active
+                      XcodeSimulatorService.app(
+                        XcodeSimulatorAppAction.terminate,
+                        props.application.bundleIdentifier,
+                        props.application.simulator
+                      ).then();
+                      try {
+                        await XcodeSimulatorService.deleteAppFiles(
+                          props.application.sandBoxPath,
+                          props.application.appGroupPath
+                        );
+                        await showToast({
+                          title: "App data deleted",
+                          message: `${props.application.name} app data has been deleted`,
+                        });
+                      } catch (error: any) {
+                        await showToast({
+                          title: "Error Deleting app data",
+                          style: Toast.Style.Failure,
+                          message: error.message,
+                        });
+                      }
```

Comment:
If you like you can make use of the [`operationWithUserFeedback`](https://github.com/raycast/extensions/blob/dd9fea8e998eb55b34bb0ea725ad8e9595e2221b/extensions/xcode/src/shared/operation-with-user-feedback.ts#L21-L33) function to add a loading indicator if the process of deletion take some time.
https://github.com/raycast/extensions/blob/dd9fea8e998eb55b34bb0ea725ad8e9595e2221b/extensions/xcode/src/shared/operation-with-user-feedback.ts#L21-L33

Reply:
Updated. Please check.